.github/workflows/main.yml

```diff
@@ -14,108 +14,62 @@ on:
 
 # A workflow run is made up of one or more jobs that can run sequentially
 jobs:
-  # This workflow contains a single job called "clang-format"
-  clang-format:
-    runs-on: self-hosted
+  build-test:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        repos: [ r3broot, sofia, frs, asyeos ]
+        include:
+          - repos: r3broot
+            cache: r3broot
+          - repos: sofia
+            url: https://github.com/R3BRootGroup/sofia.git
+            cache: sofia
+          - repos: frs
+            url: https://github.com/R3BRootGroup/frs.git
+            cache: r3broot
+          - repos: asyeos
+            url: https://github.com/R3BRootGroup/asyeos.git
+            cache: r3broot
```

Comment:
@jose-luis-rs In the future, if you wish to add more repos, you can specify them here with the corresponding url. If they are very different from the base R3BRoot, like sofia here, it could take a long time for them to build. In such case, they need their own cache to speed up the building. If not, like `frs` here, we can just use r3broot base cache for the building.

Also be very careful about the additional ctest in those repos. If those tests need some geo files, we should also create those geo files with the file `R3BRoot/uil/generate_geo_test.sh`

Reply:
Thanks for the comments!